app/models/change_service_event.rb

```diff
@@ -2,6 +2,7 @@ class ChangeServiceEvent < AggregatedEvent
   belongs_to_aggregate :service
   data_attributes :attributes_changed
   validate :validate_changes
+  after_save :publish_changes
```

Comment:
```suggestion
  after_save TriggerMetadataEventCallback.publish
```

Reply:
The suggested change doesn't work... Its what I tried first.  If there is no SP or MSA component there is also no environment which causes the `TriggerMetadataEventCallback.publish` to suffer the ruby equivalent of a null pointer exception.

Reply:
you need to update it to make it work